services/parser-database/parser.py

```diff
@@ -0,0 +1,136 @@
+                id={self.id}, \
+                jurisdiction={self.jurisdiction}, \
+                last_updated={self.last_updated}, \
+                url={self.url}, \
+                shahash={self.shahash}\
+            )"
+
+
+class Article:
+    """Class for storing articles.
+    """
+    def __init__(self, number, text):
+        self.number = number
+        self.text = text
+
+
+def identify_articles(pdf_text):
+    """Identifies articles and returns a list of Article objects.
+
+    Args:
+        pdf_text (string): contains the PDF text where articles
+        are to be identified.
+
+    Returns:
+        list: article objects
+    """
+    articles = []
+    article_text = ""
+    article_count = 1
+    i = 0
+
+    while i < len(pdf_text):
+        if (pdf_text[i] == "artículo" or pdf_text[i] == "articulo") and (
+                pdf_text[i + 1] == str(article_count) + ".-"
+                or pdf_text[i + 1] == str(article_count) + "-"
+                or pdf_text[i + 1] == str(article_count) + "."):
+            logging.info("Article #" + str(article_count) + " recognized!")
+            articles.append(Article(article_count, article_text.strip()))
+            article_text = ""
+            article_count += 1
+            i += 1
+        else:
+            article_text += " " + pdf_text[i]
+            if i == len(pdf_text) - 1:
+                articles.append(Article(article_count, article_text.strip()))
+        i += 1
+    articles.pop(0)
+    return articles
+
+
+def parse_all_documents():
+    """Parses all documents that are specified on the DB"""
+    document_list = connector.get_documents_to_parse()
+    for document in document_list:
+        file_name = document["jurisdiction"] + ".pdf"
+        retriever.get_document(document["url"], file_name)
+        logging.info('File downloaded')
+        parse(document, file_name)
+
+
+# logging.info("gcloud syntax response: %s", gcloud_response)
+def has_file_changed(past_hash, file_name):
+    """Sees if the file is different.
+
+    Args:
+        past_hash (string): hash to compare from
+
+    Returns:
+        [boolean]: [if the file has changed or not]
+    """
+    hasher = hashlib.sha256()
+    with open(file_name, "rb") as pdf_file:
+        file_buffer = pdf_file.read()
+        hasher.update(file_buffer)
+        if hasher.hexdigest() == past_hash:
+            return False
+        return True
+
+
+def article_to_dictionary(article):
+    """Converts an parsed article to a dictionary and saves it
+
+    Args:
+        article (Article): [article object]
+    """
+    article_dict = {
+        "articleNumber": article.number,
+        "text": article.text,
+        "wordCount": len(article.text.split())
+    }
+    connector.articles_in_memory[str(article.number)] = article_dict
+    keywords = connector.get_keywords(article.text)
+    logging.info('Article ' + str(article.number) + ' assigned keywords')
+    connector.save_keywords_in_memory(keywords, article)
```

Comment:
Is there a way to combine lines 116 and 119 into a single method in `connector.py` that could then be more easily mocked? It may become easier to do this when the `dict` transformation is a method of `Article` (as suggested above)